### ISSUE TITLE
watch not working in ruby 1.9.1

### ISSUE THREAD

**Issue by bdtomlin** [Compass/compass#48]
I keep getting an error when running the watch script on ruby 1.9.1.

I'm using version 0.8.17

> > > Compass is watching for changes. Press Ctrl-C to Stop.
> > > TypeError on line ["969"] of /System/Library/Frameworks/Ruby.framework/Versions/1.9.1/lib/ruby/1.9.1/pathname.rb: can't convert Pathname into String
> > > Run with --trace to see the full backtrace

I have traced it to the following:

Basically what's failing is 
Pathname.glob(Pathname.new('/Users/bryan/Sites/ushaf/public/src/**/*.sass'))

(fails in irb as well)
The pathname is passed in, I just used Pathname.new to test

but Pathname.new('/Users/bryan/Sites/ushaf/public/src/**/*.sass').to_s works fine.

If I change vendor/fssm/state.rb from

```
def add_glob(base, glob)
  Pathname.glob(base.join(glob)).each do |fn|
    @cache.set(fn)
  end
end
```

to:
    def add_glob(base, glob)
      Pathname.glob(base.join(glob).to_s).each do |fn|
        @cache.set(fn)
      end
    end

There is no longer a problem, but I'm not sure why.

the trace...

/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/1.9.1/pathname.rb:969:in `glob'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/state.rb:49:in`add_glob'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/state.rb:45:in `block in snapshot'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/state.rb:45:in`each'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/state.rb:45:in `snapshot'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/state.rb:37:in`recache'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/state.rb:9:in `refresh'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/backends/polling.rb:10:in`add_path'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm/monitor.rb:18:in `path'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/commands/watch_project.rb:26:in`block (2 levels) in perform'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/commands/watch_project.rb:25:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/commands/watch_project.rb:25:in`block in perform'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm.rb:19:in `call'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/vendor/fssm.rb:19:in`monitor'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/commands/watch_project.rb:24:in `perform'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/commands/base.rb:15:in`execute'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/commands/project_base.rb:19:in `execute'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/exec.rb:242:in`do_command'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/exec.rb:57:in `perform!'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/lib/compass/exec.rb:40:in`run!'
/System/Library/Frameworks/Ruby.framework/Versions/1.9/lib/ruby/gems/1.9.1/gems/chriseppstein-compass-0.8.16/bin/compass:8:in `<top (required)>'
/usr/local/ruby/bin/compass:19:in`load'
/usr/local/ruby/bin/compass:19:in `<main>'

**Comment by haarts** [comment]
I also encountered this problem. The fix works but it is also unclear to me why. I believe it might have something to do with the new Dir.glob method in 1.9

**Comment by chriseppstein** [comment]
Fix a bug in FSSM for ruby 1.9. Closed by bad615ef09ffe4cdb18aece31b82d57b0f2e2130.
